core/src/main/java/org/bitcoinj/core/TransactionInput.java

```diff
@@ -115,40 +115,45 @@ public static TransactionInput read(ByteBuffer payload, Transaction parentTransa
         TransactionOutPoint outpoint = TransactionOutPoint.read(payload);
         byte[] scriptBytes = Buffers.readLengthPrefixedBytes(payload);
         long sequence = ByteUtils.readUint32(payload);
-        return new TransactionInput(parentTransaction, scriptBytes, outpoint, sequence, null);
+        return new TransactionInput(parentTransaction, null, scriptBytes, outpoint, sequence, null, null);
     }
 
     public TransactionInput(@Nullable Transaction parentTransaction, byte[] scriptBytes,
                             TransactionOutPoint outpoint) {
-        this(parentTransaction, scriptBytes, outpoint, NO_SEQUENCE, null);
+        this(parentTransaction, null, scriptBytes, outpoint, NO_SEQUENCE, null, null);
     }
 
     public TransactionInput(@Nullable Transaction parentTransaction, byte[] scriptBytes,
                             TransactionOutPoint outpoint, @Nullable Coin value) {
-        this(parentTransaction, scriptBytes, outpoint, NO_SEQUENCE, value);
+        this(parentTransaction, null, scriptBytes, outpoint, NO_SEQUENCE, value, null);
     }
 
-    private TransactionInput(@Nullable Transaction parentTransaction, byte[] scriptBytes,
-                            TransactionOutPoint outpoint, long sequence, @Nullable Coin value) {
+    /** used internally */
+    public TransactionInput(@Nullable Transaction parentTransaction, @Nullable Script scriptSig, byte[] scriptBytes,
+                            TransactionOutPoint outpoint, long sequence, @Nullable Coin value,
+                            @Nullable TransactionWitness witness) {
```

Comment:
I suggest making this 7-arg constructor private and add a 5-arg constructor specifically for WalletProtobufSerializer and mark it as `/** internal use only **/`.

(We could even make the 5-arg constructor private and then put a static method in an internal-only util class if we really wanted to lock it down.)

Reply:
Changed: https://github.com/bitcoinj/bitcoinj/compare/6ee35cf0b92f935d89ec8ea6a6945085d157b398..091a578e3ae22f7aafd331505d8b1d840eba18ca